src/components/EventCardCollection.tsx

```diff
@@ -31,16 +33,49 @@ const GET_EVENTS = gql`
 const EventCardCollection = () => {
   //useQuery hook takes our gql query and makes a request to the backend
   //the return value is held in data
+  const [isModalOpen, setModalOpen] = useState(false);
+  const [modalContent, setModalContent] = useState({});
+
+  const toggleModal = () => {
+    setModalOpen(!isModalOpen);
+  };
+
+  const closeModal = () => {
+    setModalOpen(false);
+  };
+
+  const addModalContent = (
+    title: string,
```

Comment:
this is all duplication of state.  Single source of truth would be better.  You have a list of events, all you need to store in state is which is the selected event.   Then render it's data in the modal